### PR TITLE
test(MCDA): 19182 more complex unit tests for layerCalculationPipeline

### PR DESCRIPTION
**Pull request by albaranau**
https://kontur.fibery.io/Tasks/Task/Unit-tests-for-MCDA-transformations-19182

## Summary by CodeRabbit

- **New Features**
  - Introduced enhanced data visualization capabilities with D3 Scale and Recharts libraries.
  - Added a new function for axis transformations in the MCDA API.
  - Enhanced MCDA Layer Editor with new transformation controls and interactivity improvements.
  - Improved data handling capabilities with a new dependency for converting CSV data to JSON.
  - Introduced new styling for transformation selection rows to enhance user experience.
  - Added a new optional callback for double-clicking titles in the MCDA Layer Parameter Row.

- **Documentation**
  - Added a comprehensive README file outlining the testing framework for the calculateLayerPipeline function, detailing data usage, updating processes, and the implementation of a clamp function.

- **Style**
  - Refactored CSS for various components to improve layout and usability, including the addition of new styling classes.

- **Tests**
  - Added comprehensive test coverage for new calculations and transformations in the MCDA framework.

- **Chores**
  - Cleaned up unnecessary dependencies and optimized the project's configuration for better performance.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -136,6 +136,7 @@
     "bundle-size-diff": "github:konturio/bundle-size-diff#v1.0.7",
     "clsx": "^1.2.1",
     "cpr": "^3.0.1",
+    "csvtojson": "^2.0.10",
```
Comment (vkozio) [inline package.json]: we already have **papaparse** in deps

**Review by vkozio — CHANGES_REQUESTED** [review]
Please reuse lib from deps

File: package.json (hunk 1 above)
Comment (albaranau) [reply package.json]: Fixed

**Review by vkozio — APPROVED** [review]

File: src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/testData/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+The test uses `calculateLayerPipeline.testdata.csv` file to get the input data and expected values. This is a static CSV file exported from `calculateLayerPipeline.formulas.xlsx`.
+
+## Test data files
+
+- `calculateLayerPipeline.testdata.csv` is a plain CSV file. Headers are using dot notation (parent.child.grandchild) to create nested objects in tests.
+
+- `calculateLayerPipeline.formulas.xlsx` contains two sheets:
+  - `mcda_calc` has actual calculations and formulas
+  - `mcdaCalculations.testdata` imports necessary fields from `mcda_calc`
```
[the remark this answers is not in the document]
Comment (albaranau) [reply src/core/logical_layers/renderers/stylesConfigs/mcda/calculations/testData/readme.md]: Should be fixed

**Pull request merged**